nkms/crypto/api.py

```diff
@@ -67,378 +57,43 @@ def keccak_digest(
-    :param min_shares: Minimum shares needed to reproduce rekey
-    :param total_shares: Total shares to generate from split-rekey gen
-
-    :return: A list of RekeyFrags to distribute
-    """
-    if type(privkey_a) == bytes:
-        privkey_a = priv_bytes2ec(privkey_a)
-    if type(privkey_b) == bytes:
-        privkey_b = priv_bytes2ec(privkey_b)
-    umbral_rekeys = PRE.split_rekey(privkey_a, privkey_b,
-                                    min_shares, total_shares)
-    return [KFrag(umbral_kfrag=u) for u in umbral_rekeys]
-
-
-def ecies_ephemeral_split_rekey(
-        privkey_a: Union[bytes, elliptic_curve.ec_element],
-        pubkey_b: Union[bytes, elliptic_curve.ec_element],
-        min_shares: int,
-        total_shares: int
-) -> Tuple[List[umbral.RekeyFrag], Tuple[bytes, bytes]]:
-    """
-    Performs a split-key re-encryption key generation where a minimum
-    number of shares `min_shares` are required to reproduce a rekey.
-    Will split a rekey inot `total_shares`.
-    This also generates an ephemeral keypair for the recipient as `pubkey_b`.
-
-    :param privkey_a: Privkey to re-encrypt from
-    :param pubkey_b: Public key to re-encrypt for (w/ ephemeral key)
-    :param min_shares: Minium shares needed to reproduce a rekey
-    :param total_shares: Total shares to generate from split-rekey gen
-
-    :return: A tuple containing a list of rekey frags, and a tuple of the
-             encrypted ephemeral key data (enc_symm_key, enc_eph_privkey)
-    """
-    eph_privkey, (encrypted_key, encrypted_message) = _internal._ecies_gen_ephemeral_key(pubkey_b)
-    kfrags = ecies_split_rekey(privkey_a, eph_privkey, min_shares, total_shares)
-    pfrag = PFrag(ephemeral_data_as_bytes=None, encrypted_key=encrypted_key, encrypted_message=encrypted_message)
-
-    return (kfrags, pfrag)
-
-
-def ecies_combine(
-        cfrags: List[CFrag]
-) -> umbral.EncryptedKey:
-    """
-    Combines the encrypted keys together to form a rekey from split_rekey.
-
-    :param encrypted_keys: Encrypted keys to combine
-
-    :return: The combined EncryptedKey of the rekey
-    """
-    return PRE.combine([cfrag.encrypted_key for cfrag in cfrags])
-
-
-def ecies_reencrypt(
-        rekey: Union[bytes, umbral.RekeyFrag],
-        enc_key: Union[bytes, umbral.EncryptedKey],
-) -> umbral.EncryptedKey:
-    """
-    Re-encrypts the key provided.
-
-    :param rekey: Re-encryption key to use
-    :param enc_key: Encrypted key to re-encrypt
-
-    :return: The re-encrypted key
-    """
-    if type(rekey) == bytes:
-        rekey = umbral.RekeyFrag(None, priv_bytes2ec(rekey))
-    if type(enc_key) == bytes:
-        enc_key = umbral.EncryptedKey(priv_bytes2ec(enc_key), None)
-    reencrypted_data = PRE.reencrypt(rekey, enc_key)
-    return CFrag(reencrypted_data=reencrypted_data)
+    try:
+        cryptography_pub_key.verify(
+            signature,
+            message,
+            ec.ECDSA(hashes.BLAKE2b(64))
```

Comment:
I wonder if we should have `hashes.BLAKE2b(64)` like a constant somewhere

Reply:
👍